apps/google-analytics-4/src/components/FormControlServiceAccountKey.tsx

```diff
@@ -186,33 +188,37 @@ const FormControlServiceAccountKey = ({
   };
 
   const formControl = (
-    <FormControl marginTop="spacingL" id="accountCredentialsFile">
-      <FormControl.Label>Google Service Account Key File</FormControl.Label>
+    <FormControl
+      marginTop="spacingL"
+      id="accountCredentialsFile"
+      isInvalid={formControlState.isInvalid}
+      isRequired={isRequired}
+    >
+      <FormControl.Label isRequired={isRequired}>Google Service Account Key File</FormControl.Label>
       <Textarea
         name="accountCredentialsFile"
         placeholder={placeholderText}
         rows={10}
         className={styles.credentialsInput}
         value={serviceAccountKeyFile}
         onChange={onKeyFileTextareaChange}
-        isInvalid={formControlState.isInvalid}
       />
       {formControlState.isInvalid ? (
         <FormControl.ValidationMessage>
           Error: {formControlState.errorMessage}
         </FormControl.ValidationMessage>
+      ) : serviceAccountKeyFile ? (
+        <Flex marginTop="spacingXs" alignItems="center">
+          <Flex isInline={true} alignItems="center">
```

Comment:
Double flex? What are you doing here?

Reply:
ahh yes, sorry this is a bit weird. basically i ended up using the lower level HTML from the component `FormControl.ValidationMessage`. More or less recreating this component: https://github.com/contentful/forma-36/blob/main/packages/components/forms/src/ValidationMessage/ValidationMessage.tsx#L40-L45

The reason being that I wanted a component that's more or less identical to the error to show a "valid" state, with the same height and style etc., just a different icon. The built in validation message is only for errors.

It's pretty much the same nesting of Flex elements they do there. Is there a better way, do you think?

Reply:
Okay, it's fine then. I should probably ask the same question about double Flex in the F36 repo :D